src/components/Lightbox.tsx

```diff
@@ -1,78 +1,84 @@
-/* eslint-disable es/no-optional-chaining */
-import PropTypes from 'prop-types';
 import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import type {LayoutChangeEvent, NativeSyntheticEvent, StyleProp, ViewStyle} from 'react-native';
 import {ActivityIndicator, PixelRatio, StyleSheet, View} from 'react-native';
 import useStyleUtils from '@hooks/useStyleUtils';
-import * as AttachmentsPropTypes from './Attachments/propTypes';
 import Image from './Image';
-import MultiGestureCanvas from './MultiGestureCanvas';
+import MultiGestureCanvas, {defaultZoomRange} from './MultiGestureCanvas';
 import getCanvasFitScale from './MultiGestureCanvas/getCanvasFitScale';
-import {zoomRangeDefaultProps, zoomRangePropTypes} from './MultiGestureCanvas/propTypes';
+import type {ContentSize, OnScaleChangedCallback, ZoomRange} from './MultiGestureCanvas/types';
 
 // Increase/decrease this number to change the number of concurrent lightboxes
 // The more concurrent lighboxes, the worse performance gets (especially on low-end devices)
-// -1 means unlimited
-const NUMBER_OF_CONCURRENT_LIGHTBOXES = 3;
+type LightboxConcurrencyLimit = number | 'UNLIMITED';
+const NUMBER_OF_CONCURRENT_LIGHTBOXES: LightboxConcurrencyLimit = 3;
+const DEFAULT_IMAGE_SIZE = 200;
+const DEFAULT_IMAGE_DIMENSION: ContentSize = {width: DEFAULT_IMAGE_SIZE, height: DEFAULT_IMAGE_SIZE};
 
-const cachedDimensions = new Map();
+type LightboxImageDimensions = {
+    lightboxSize?: ContentSize;
+    fallbackSize?: ContentSize;
+};
 
-/**
- * On the native layer, we use a image library to handle zoom functionality
- */
-const propTypes = {
-    ...zoomRangePropTypes,
+const cachedDimensions = new Map<string, LightboxImageDimensions>();
 
-    /** Function for handle on press */
-    onPress: PropTypes.func,
+type ImageOnLoadEvent = NativeSyntheticEvent<ContentSize>;
 
-    /** Handles errors while displaying the image */
-    onError: PropTypes.func,
+type LightboxProps = {
+    /** Whether source url requires authentication */
+    isAuthTokenRequired: boolean;
 
-    /** URL to full-sized attachment, SVG function, or numeric static image on native platforms */
-    source: AttachmentsPropTypes.attachmentSourcePropType.isRequired,
+    /** URI to full-sized attachment, SVG function, or numeric static image on native platforms */
+    uri: string;
 
-    /** Whether source url requires authentication */
-    isAuthTokenRequired: PropTypes.bool,
+    /** Triggers whenever the zoom scale changes */
+    onScaleChanged?: OnScaleChangedCallback;
 
-    /** Whether the Lightbox is used within a carousel component and there are other sibling elements */
-    hasSiblingCarouselItems: PropTypes.bool,
+    /** Handles errors while displaying the image */
+    onError: () => void;
+
+    /** Additional styles to add to the component */
+    style: StyleProp<ViewStyle>;
```

Comment:
```suggestion
    style?: StyleProp<ViewStyle>;
```

Style props are usually optional.